pkg/costmodel/allocation.go

```diff
@@ -954,6 +960,33 @@ func applyGPUsAllocated(podMap map[podKey]*Pod, resGPUsRequested []*prom.QueryRe
 
 		hrs := pod.Allocations[container].Minutes() / 60.0
 		pod.Allocations[container].GPUHours = res.Values[0].Value * hrs
+		pod.Allocations[container].GPURequestAverage = res.Values[0].Value
```

Comment:
Is this an actual average? Should we be querying this more like CPU/RAM request averages?

Reply:
It is. Can you expand on this? To my knowledge it is being queried and handled like RAM requests.

Reply:
This function does some funky stuff with `queryFmtGPUsAllocated` -- look at the first `if` statement in this function, I think you _might_ run into trouble because requests is a separate array and can get overwritten by allocation.

However, I'm not super familiar with GPU. This might not be a problem at all, in which case :+1: 

Other resources (RAM, CPU, etc) have separate function for applying requests, usage, and average. We don't need to adhere to that style if this is correct, its just what tipped me off that this looked a little odd.

Reply:
Ah, yes I see. I believe this is consistent with what we do throughout, which is override requests with limits (which should also be what is happening in any pod using a GPU regardless; you're just supposed to have limits).

There may be problems with this particular function that I am not aware of, but the value of "GPUs allocated per allocation" right now is back-calculated from `GPUHours` anyway, so really all this change does is make that a separate field in Allocation.